### ISSUE TITLE
Add types for subscriptions config

### ISSUE THREAD

**Issue by nguyen-vo** [terraform-google-modules/terraform-google-pubsub#172]
### TL;DR

Add type to the subscription configs. This helps improve developer experience and ease to use

### Terraform Resources

```markdown
terraform-google-modules/terraform-google-pubsub
```

### Additional information
We have different subscription configs, but currently, they have any type that is hard for new developers to use the module. I had to spend sometimes jump back and forth between the PubSub resource page on the GCP provider Terraform page and GitHub to figure out how to set up BigQuery subscription with other attributes like ack_deadline_seconds, dead_letter_policy, etc..

The BigQuery module has a great table for the input https://github.com/terraform-google-modules/terraform-google-bigquery/blob/master/README.md

Greatly appreciate if we could add something like it

_No response_
